src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -100,13 +100,14 @@ public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
             var isKey = Transformer.ValueColumnType.IsKey;
             var columnType = (isKey) ? PrimitiveType.FromKind(DataKind.U4) :
                                     Transformer.ValueColumnType;
+            var metadata = SchemaShape.Create(Transformer.ValueColumnMetadata.Schema);
             foreach (var (Input, Output) in _columns)
             {
                 if (!inputSchema.TryFindColumn(Input, out var originalColumn))
                     throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", Input);
 
                 // Get the type from TOutputType
-                var col = new SchemaShape.Column(Output, vectorKind, columnType, isKey, originalColumn.Metadata);
+                var col = new SchemaShape.Column(Output, vectorKind, columnType, isKey, metadata);
```

Comment:
Although Name is more specific to what the object actually is, I think it goes against thinking of these as columns since you are talking about a input/output (or source/name). I would prefer to keep it as Output/Input. BTW - this will also go through another revision with #2064

---
In reply to: [247282892](https://github.com/dotnet/machinelearning/pull/2098#discussion_r247282892) [](ancestors = 247282892)